src/main/java/edu/cornell/kfs/pdp/batch/service/impl/CuAchAdviceNotificationErrorReportServiceImpl.java

```diff
@@ -0,0 +1,138 @@
+        reportWriterService.setTitle(configurationService.getPropertyValueAsString(CUPdpKeyConstants.PDP_SEND_ACH_NOTIFICATION_ERROR_REPORT_TITLE));
+        reportWriterService.initialize();
+        reportWriterService.writeNewLines(2);
+        String rowFormat = configurationService.getPropertyValueAsString(CUPdpKeyConstants.PDP_SEND_ACH_NOTIFICATION_ERROR_REPORT_FORMAT);
+        reportWriterService.writeFormattedMessageLine(rowFormat, "PAYEE ID", "PAYMENT GROUP", "DISBURSEMENT NUMBER", "EMAIL ADDRESS");
+        String hyphens25 = buildHyphenString(20);
+        String hyphens50 = buildHyphenString(50);
+        reportWriterService.writeFormattedMessageLine(rowFormat, hyphens25, hyphens25, hyphens25, hyphens50);
+    }
+    
+    private String buildHyphenString(int numberOfHyphens) {
+        StringBuilder sb = new StringBuilder();
+        for (int i=0; i<numberOfHyphens; i++) {
+            sb.append("-");
+        }
+        return sb.toString();
+    }
+    
+    private void printErrorReportDetails(List<PDPBadEmailRecord> badEmailRecords) {
+        String rowFormat = configurationService.getPropertyValueAsString(CUPdpKeyConstants.PDP_SEND_ACH_NOTIFICATION_ERROR_REPORT_FORMAT);
+        for (PDPBadEmailRecord record : badEmailRecords) {
+            reportWriterService.writeFormattedMessageLine(rowFormat, record.getPayeeId(), record.getPaymentGroupId(), record.getDisbursementNumber(), record.getEmailAddress());
+        }
+        reportWriterService.writeNewLines(1);
+    }
+
+    @Override
+    public void emailBadEmailReport(File errorReport) {
+        if (errorReport != null) {
+            String errorReportContents = concurBatchUtilityService.getFileContents(errorReport.getAbsolutePath());
+            String fromAddress = findPdpFromEmailAddress();
+            List<String> toAddressList = findPDPToEmailAddress();
+            String subject = configurationService.getPropertyValueAsString(CUPdpKeyConstants.PDP_SEND_ACH_NOTIFICATION_ERROR_REPORT_TITLE);
+            
+            BodyMailMessage message = new BodyMailMessage();
+            message.setFromAddress(fromAddress);
+            message.setSubject(subject);
+            message.getToAddresses().addAll(toAddressList);
+            message.setMessage(errorReportContents);
+            
+            try {
+                emailService.sendMessage(message, false);
+            } catch (Exception e) {
+                LOG.error("emailBadEmailReport, the email could not be sent", e);
+            }
+            
+        } else {
+            LOG.error("emailBadEmailReport, no error report sent in");
+        }
+
+    }
+    
+    private String findPdpFromEmailAddress() {
+        return parameterService.getParameterValueAsString(KFSConstants.ParameterNamespaces.PDP, KfsParameterConstants.BATCH_COMPONENT, KFSConstants.FROM_EMAIL_ADDRESS_PARM_NM);
```

Comment:
This one is still KFSContstants.ParameterNamespaces.PDP

Reply:
opps